pkg/apis/logging/v1/elasticsearch_types.go

```diff
@@ -59,11 +60,32 @@ type ElasticsearchSpec struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	ManagementState  ManagementState       `json:"managementState"`
-	RedundancyPolicy RedundancyPolicyType  `json:"redundancyPolicy"`
-	Nodes            []ElasticsearchNode   `json:"nodes"`
-	Spec             ElasticsearchNodeSpec `json:"nodeSpec"`
-	IndexManagement  *IndexManagementSpec  `json:"indexManagement"`
+	// Indicator if the resource is 'Managed' or 'Unmanaged' by the operator
+	//
+	// +kubebuilder:validation:Enum:=Managed;Unmanaged
+	// +optional
+	ManagementState ManagementState `json:"managementState"`
+
+	// The policy towards data redundancy to specify the number of redundant primary shards
+	//
+	// +optional
```

Comment:
not optional

Reply:
can we also add an enum for this?

Reply:
This is already generated as an enum based on the golang type and the predefined constants.

Reply:
ack, it just looked odd to not have it explicit here like `+kubebuilder:validation:Enum:=` like in other place :)